### PR TITLE
Fix for device registration regression

### PR DESCRIPTION
**Pull request by lpusok**
### Checklist
- [x] I've read and followed the [Contribution Guidelines](https://github.com/bitrise-steplib/.github/blob/main/CONTRIBUTING.md)
- [x] `step.yml` and `README.md` is updated with the changes (if needed)

### Version
Requires a *PATCH* [version update](https://semver.org/)

### Context

errors.As was incorrectly used:
```
	rerr := appstoreconnect.ErrorResponse{} <-- & missing
		if ok := errors.As(err, &rerr); ok {
```

Resolves: https://bitrise.atlassian.net/browse/STEP-1731

### Changes

- Correctly detect  appstoreconnect.ErrorResponse.
- Fixed a bug when an incorrect device is encountered, other devices still sill be registered instead of returning early.
- Added retry for fetching a Bundle ID from a profile, if the profile has been removed.

### CODE REVIEW

File: autocodesign/devportalclient/appstoreconnectclient/devices.go [hunk 1]
```diff
@@ -64,7 +64,7 @@ func (d *DeviceClient) RegisterDevice(testDevice devportalservice.TestDevice) (*
 
 	registeredDevice, err := d.client.Provisioning.RegisterNewDevice(req)
 	if err != nil {
-		rerr := appstoreconnect.ErrorResponse{}
+		rerr := &appstoreconnect.ErrorResponse{}
```
Comment (tothszabi) [inline autocodesign/devportalclient/appstoreconnectclient/devices.go:67]: In the `profiles.go` file you were using the 
```
var respErr *appstoreconnect.ErrorResponse
```
format at both places and I wonder if we could use the same here as well just for consistency.

File: autocodesign/devportalclient/appstoreconnectclient/profiles_test.go [hunk 2]
```diff
@@ -191,3 +190,36 @@ func newResponse(t *testing.T, status int, body map[string]interface{}) *http.Re
...
+					StatusCode: http.StatusNotFound,
+				},
+			},
+			wantErrType: autocodesign.ProfilesInconsistentError{},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			err := wrapInProfileError(tt.err)
+			if tt.wantErrType == nil {
```
Comment (tothszabi) [inline autocodesign/devportalclient/appstoreconnectclient/profiles_test.go:218]: Do we need this if logic here if none of the test cases a nil `wantErrType` value? I would remove it and later add it back if it will be ever needed.

**Review by tothszabi — APPROVED** [review]

**Review by godrei — APPROVED** [review]
